pydrive2/auth.py

```diff
@@ -348,24 +346,13 @@ def LoadCredentialsFile(self, credentials_file=None):
         try:
             storage = Storage(credentials_file)
             self.credentials = storage.get()
+        except KeyError:
```

Comment:
I really couldn't find a way out of this. It tries to load credentials from the saved backend, though the saved backend actually doesn't have the proper shape (missing fields like `_module`/`_class` that normally get added during serialization). Any idea @shcheklein?

Reply:
qq - so, we have a JSON file that is the one that defines the service account, and we have the credentials file - I thought that those two are conceptually different files (one is to avoid auth and defines the account), second (backend) is to store OAuth tokens. Is it a correct assumption @isidentical ? 

In this case we, this code should not be loading the JSON file that we get for the service account.

Reply:
So there is one file (the original location) that we use to authenticate in the first place and then put the storage. These 2 have a different structure, one is the raw thing that you get from the GCP console and the second is the literally the deserialized credentials instance. The problem is that, the decorator tries to load the file though it expects `_module`/`_class` from that field while that file shouldn't contain any of these.

Reply:
@isidentical thanks!

Just to clarify:

> in the first place and then put the storage

it creates the second file, right? the one that is `.dvc/tmp/gdrive-credentials.json` or something. That has a refresh token, etc?

> The problem is that, the decorator tries to load the file

which one of the files? The way I understand this is that it should not be trying to do this when there is no file (credentials file in this case `.dvc/tmp/gdrive-credentials.json`) at all - it should be creating an empty instance. (that's guess how it worked with p12, or am I wrong?)

> from that field while that file shouldn't contain any of these.

same question, which one of the files?

Reply:
> it creates the second file, right? the one that is .dvc/tmp/gdrive-credentials.json or something. That has a refresh token, etc?

yeah

> which one of the files? The way I understand this is that it should not be trying to do this when there is no file (credentials file in this case .dvc/tmp/gdrive-credentials.json) at all - it should be creating an empty instance. (that's guess how it worked with p12, or am I wrong?)

yeah, though there is the file 

So we have a `GDRIVE_BLAH_BLAH` secret and it gets written into `credentials/default.dat`;
https://github.com/iterative/PyDrive2/blob/e09b9d8e54f8db03129efa3727f1109e3ee277e3/pydrive2/test/test_util.py#L21-L27

This is the raw file obtained from GCP, not the file that is generated after logging in. So this is not the `.dvc/tmp/gdrive-credentials.json` file. We put this file to `credentials/default.dat` and try to run our tests. Though the `settings/default.yaml` claims that file is the path where we would save the auth results (like `.dvc/tmp/gdrive-credentials.json`) and not the origin. 
https://github.com/iterative/PyDrive2/blob/e09b9d8e54f8db03129efa3727f1109e3ee277e3/pydrive2/test/settings/default.yaml#L7

And the origin is some path that doesn't exist (I also had to change it in this PR); 
https://github.com/iterative/PyDrive2/blob/e09b9d8e54f8db03129efa3727f1109e3ee277e3/pydrive2/test/settings/default.yaml#L3

This is the weird part that why those worked under the p12 auth but not on json

Reply:
Can it be that in case of P12 we had an actual credentials file serialized in the GDRIVE_USER_CREDENTIALS_DATA on CI? And in this case I mad a mistake and serialized the raw file?

Reply:
if it can find the credentials file (and assuming refresh token is valid for a long time) it probably doesn't even try to go with p12 and/or JSON ... which breaks the point of this test to some extent, but it's a different story

Reply:
> Can it be that in case of P12 we had an actual credentials file serialized in the GDRIVE_USER_CREDENTIALS_DATA on CI?

that might be the odd part

Reply:
yeah, but from I what I see this the only reasonable explanation... and this is the only auth test that we run automatically (all other auth tests are manual (for a good reason)). We can create the GDRIVE_SERVICE_ACCOUNT_JSON, omit setup credentials call, write this var into the `your_file_...json` for this test. That looks like a proper fix for this.

Reply:
@shcheklein could you please create `GDRIVE_SERVICE_ACCOUNT_JSON` with the same contents of `GDRIVE_USER_CREDENTIALS_DATA` and I'll adapt this patch

Reply:
Nevermind the last comment